src/java/org/apache/cassandra/db/memtable/TrieMemtable.java

```diff
@@ -246,15 +246,22 @@ public long put(PartitionUpdate update, UpdateTransaction indexer, OpOrder.Group
         MemtableShard shard = shards[boundaries.getShardForKey(key)];
         long colUpdateTimeDelta = shard.put(update, indexer, opGroup);
 
-        if (shard.data.reachedAllocatedSizeThreshold() && !switchRequested.getAndSet(true))
-        {
-            logger.info("Scheduling flush due to trie size limit reached.");
-            owner.signalFlushRequired(this, ColumnFamilyStore.FlushReason.MEMTABLE_LIMIT);
-        }
+        if (shard.data.reachedAllocatedSizeThreshold())
+            signalFlushRequired(ColumnFamilyStore.FlushReason.TRIE_LIMIT);
 
         return colUpdateTimeDelta;
     }
 
+    @Override
+    public void signalFlushRequired(ColumnFamilyStore.FlushReason flushReason)
+    {
+        if (!switchRequested.getAndSet(true))
+        {
+            logger.info("Scheduling flush for table {} due to {}", this.metadata.get(), flushReason);
```

Comment:
Nit: It may be valuable to see all the reasons a flush was requested, not just the first.